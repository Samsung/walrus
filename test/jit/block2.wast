(module
(func $test1 (local i32)
  block
    i32.const 1
    if (result i32)
      i32.const 2
    else
      i32.const 3
      if (result i32)
        i32.const 2
      else
        i32.const 2
        i32.const 5
        br_if 1
      end

      local.set 0
      br 1
    end

    local.set 0
  end
)

(func $test2 (local i32)
  block
    i32.const 1
    if (result i32)
      local.get 0
    else
      i32.const 3
      if (result i32)
        local.get 0
      else
        local.get 0
        i32.const 5
        br_if 1
      end

      local.set 0
      br 1
    end

    local.set 0
  end
)

(func $test3 (local i32)
  block
    i32.const 1
    if (result i32)
      i32.const 2
    else
      i32.const 3
      if (result i32)
        local.get 0
      else
        local.get 0
        i32.const 5
        br_if 1
      end

      local.set 0
      br 1
    end

    local.set 0
  end
)

(func $test4 (local i32 i32)
  block
    i32.const 1
    if (result i32)
      i32.const 2
    else
      i32.const 3
      if (result i32)
        local.get 0
      else
        local.get 0
        i32.const 5
        br_if 1
      end

      local.set 0
      br 1
    end

    local.set 1
  end
)
)
