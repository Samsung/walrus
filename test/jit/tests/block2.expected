[[[[[[[  Function   0  ]]]]]]]
Local 0: mapped to: 0 range [14 - 17]
Number of locals reduced from 1 to 1
0: Opcode: i32.const
  Result[0]: i32
1: Opcode: br_unless
  Jump to: 4
  Param[0]: ref:0(0)
2: Opcode: i32.const
  Result[0]: i32
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32
6: Opcode: br_unless
  Jump to: 9
  Param[0]: ref:5(0)
7: Opcode: i32.const
  Result[0]: i32
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: i32.const
  Result[0]: i32
11: Opcode: i32.const
  Result[0]: i32
12: Opcode: br_if
  Jump to: 16
  Param[0]: ref:11(0)
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Index: 0
  Param[0]: ref:13(0)
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Index: 0
  Param[0]: ref:16(0)
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: i32.const
  Result[0]: i32 unused
1: Opcode: br_unless
  Jump to: 4
  Param[0]: i32 const:0
2: Opcode: i32.const
  Result[0]: i32 unused
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32 unused
6: Opcode: br_unless
  Jump to: 9
  Param[0]: i32 const:5
7: Opcode: i32.const
  Result[0]: i32 unused
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: i32.const
  Result[0]: i32 unused
11: Opcode: i32.const
  Result[0]: i32 unused
12: Opcode: br_if
  Jump to: 16
  Param[0]: i32 const:11
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 const:7
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 const:2
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return

[[[[[[[  Function   1  ]]]]]]]
Local 0: mapped to: 0 range [BEGIN - 17]
Number of locals reduced from 1 to 1
0: Opcode: i32.const
  Result[0]: i32
1: Opcode: br_unless
  Jump to: 4
  Param[0]: ref:0(0)
2: Opcode: local.get
  Index: 0
  Result[0]: i32
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32
6: Opcode: br_unless
  Jump to: 9
  Param[0]: ref:5(0)
7: Opcode: local.get
  Index: 0
  Result[0]: i32
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: local.get
  Index: 0
  Result[0]: i32
11: Opcode: i32.const
  Result[0]: i32
12: Opcode: br_if
  Jump to: 16
  Param[0]: ref:11(0)
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Index: 0
  Param[0]: ref:13(0)
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Index: 0
  Param[0]: ref:16(0)
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: i32.const
  Result[0]: i32 unused
1: Opcode: br_unless
  Jump to: 4
  Param[0]: i32 const:0
2: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 unused
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32 unused
6: Opcode: br_unless
  Jump to: 9
  Param[0]: i32 const:5
7: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 unused
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 unused
11: Opcode: i32.const
  Result[0]: i32 unused
12: Opcode: br_if
  Jump to: 16
  Param[0]: i32 const:11
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 sp[0]
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 sp[0]
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return

[[[[[[[  Function   2  ]]]]]]]
Local 0: mapped to: 0 range [BEGIN - 17]
Number of locals reduced from 1 to 1
0: Opcode: i32.const
  Result[0]: i32
1: Opcode: br_unless
  Jump to: 4
  Param[0]: ref:0(0)
2: Opcode: i32.const
  Result[0]: i32
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32
6: Opcode: br_unless
  Jump to: 9
  Param[0]: ref:5(0)
7: Opcode: local.get
  Index: 0
  Result[0]: i32
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: local.get
  Index: 0
  Result[0]: i32
11: Opcode: i32.const
  Result[0]: i32
12: Opcode: br_if
  Jump to: 16
  Param[0]: ref:11(0)
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Index: 0
  Param[0]: ref:13(0)
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Index: 0
  Param[0]: ref:16(0)
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: i32.const
  Result[0]: i32 unused
1: Opcode: br_unless
  Jump to: 4
  Param[0]: i32 const:0
2: Opcode: i32.const
  Result[0]: i32 sp[0]
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32 unused
6: Opcode: br_unless
  Jump to: 9
  Param[0]: i32 const:5
7: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 sp[0]
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 sp[0]
11: Opcode: i32.const
  Result[0]: i32 unused
12: Opcode: br_if
  Jump to: 16
  Param[0]: i32 const:11
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 unused
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 unused
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return

[[[[[[[  Function   3  ]]]]]]]
Local 0: mapped to: 0 range [BEGIN - 14]
Local 1: mapped to: 0 range [17 - 17]
Number of locals reduced from 2 to 1
0: Opcode: i32.const
  Result[0]: i32
1: Opcode: br_unless
  Jump to: 4
  Param[0]: ref:0(0)
2: Opcode: i32.const
  Result[0]: i32
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32
6: Opcode: br_unless
  Jump to: 9
  Param[0]: ref:5(0)
7: Opcode: local.get
  Index: 0
  Result[0]: i32
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: local.get
  Index: 0
  Result[0]: i32
11: Opcode: i32.const
  Result[0]: i32
12: Opcode: br_if
  Jump to: 16
  Param[0]: ref:11(0)
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Index: 0
  Param[0]: ref:13(0)
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Index: 0
  Param[0]: ref:16(0)
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: i32.const
  Result[0]: i32 unused
1: Opcode: br_unless
  Jump to: 4
  Param[0]: i32 const:0
2: Opcode: i32.const
  Result[0]: i32 sp[0]
3: Opcode: br
  Jump to: 16
4: Label: result_count: 0 preserved_count: 0
  Jump from: 1
5: Opcode: i32.const
  Result[0]: i32 unused
6: Opcode: br_unless
  Jump to: 9
  Param[0]: i32 const:5
7: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 sp[0]
8: Opcode: br
  Jump to: 13
9: Label: result_count: 0 preserved_count: 0
  Jump from: 6
10: Opcode: local.get
  Addr: sp[0]
  Result[0]: i32 sp[0]
11: Opcode: i32.const
  Result[0]: i32 unused
12: Opcode: br_if
  Jump to: 16
  Param[0]: i32 const:11
13: Label: result_count: 1 preserved_count: 0
  Jump from: 8
  Param[0]
    7(0) instruction
    10(0) instruction
14: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 unused
15: Opcode: br
  Jump to: 18
16: Label: result_count: 1 preserved_count: 0
  Jump from: 3
  Jump from: 12
  Param[0]
    2(0) instruction
    10(0) instruction
17: Opcode: local.set
  Addr: sp[0]
  Param[0]: i32 unused
18: Label: result_count: 0 preserved_count: 0
  Jump from: 15
19: Opcode: return

